### PR TITLE
Commons bot

### PR DESCRIPTION
**Pull request by gbirke**
Add a bot that checks a Commons category for new entries, checks the entry pages for special comments and inserts links to these pages in the German Wikipedia.

### CODE REVIEW

**Comment by WMDE-Fisch** [comment]
when running the tests I get

```
..............................................ERROR: Invalid list callback param: 'de|Test Page'
..........................
```
and travis https://travis-ci.org/wmde/WikiLovesMonuments/builds/74590329 gives
```
................................................................ERROR: Invalid list callback param: 'de|Test Page'

ERROR:pywiki:Invalid list callback param: 'de|Test Page'

......VERBOSE:pywiki:Image is already filled for id '123' on page 'Test Page', skipping ...

..
```
o.O

**Comment by gbirke** [comment]
The error output was caused by pywikibot logging writing to the output. All the logging in commons_bot tests is now mocked.

**Comment by KaiNissen** [comment]
Looks fine to me, tests pass, tested some stuff locally.

File: update-bot/wlmbots/pageinfo.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                commonscat_mapper.get_commonscat_from_weblinks_template(text),
+            "valid_id": template_checker.has_valid_id(template)
+        }
+    if info:
+        info["duplicate_ids"] = id_count > 1
+    else:
+        info["id_not_found"] = True
+    return info
+
+def main():
```
Comment (wmde-manicki) [inline update-bot/wlmbots/pageinfo.py]: Functions here on not part of class so double blank lines could be used here

File: update-bot/README.md [hunk 2]
```diff
@@ -44,6 +45,19 @@ page uses two MediaWiki templates, [`Fehler_in_Denkmallisten_Tabellenkopf`](wiki
 
 Configuration of the approved templates can be found in the file [`template_config.json`](template_config.json). It configures the names of the approved templates, the name of the ID parameter and a regular expression for validating ID parameters.
 
+
+## commons_bot.py
+Look in Wikimedia Commons in the "Wiki Loves Monuments 2015" category for comments left by the Upload Wizard. Use the comments to update the monument list pages in the German Wikipedia.
```
Comment (wmde-manicki) [inline update-bot/README.md]: For consistency with descriptions of other bots this sentence could say "Looks in..." and "Uses the comments..."

**Comment by wmde-manicki** [comment]
+1
43 commits, that's quite a pull request :)

Looked through the code again and looks fine for me now. Pointed two really minor things that maybe don't even need to be addressed right now. Just would prefer someone else to be the one that hits the button.

**Comment by gbirke** [comment]
Closing PR because of force-pushed rebase.

**Pull request closed without merging**